pkg/istio-agent/sds-agent.go

```diff
@@ -353,13 +357,20 @@ func newSecretCache(serverOptions sds.Options) (workloadSecretCache *cache.Secre
 				log.Info("Using citadel CA for SDS")
 				serverOptions.CAEndpoint = "istio-citadel.istio-system:8060"
 			} else {
-				rootCert, err = ioutil.ReadFile(k8sCAPath)
-				if err != nil {
-					log.Warnf("Failed to load K8S cert, assume IP secure network: %v", err)
-					serverOptions.CAEndpoint = "istiod.istio-system.svc:15010"
-				} else {
-					log.Info("Using default istiod CA, with K8S certificates for SDS")
+				rootCert, err = ioutil.ReadFile(CitadelCACertPath + "/" + bootstrap.CACertNamespaceConfigMapDataName)
+				if err == nil {
+					log.Info("istiod uses Citadel issued certificate")
+					log.Info("Istio Agent uses default istiod CA")
 					serverOptions.CAEndpoint = "istiod.istio-system.svc:15012"
+				} else { // istiod certificate is not issued by Citadel
+					rootCert, err = ioutil.ReadFile(k8sCAPath)
+					if err != nil {
+						log.Warnf("Failed to load K8S cert, assume IP secure network: %v", err)
+						serverOptions.CAEndpoint = "istiod.istio-system.svc:15010"
```

Comment:
given we don't have an `istiod.istio-system` I feel like someting here must be wrong?

Reply:
@howardjohn @costinm "istiod.istio-system" is from the implementation in the current master branch. I think: 1) "istiod" should be "istio-pilot" instead. 2) All hard-coded "istio-system" should be fixed. What do you think?

Reply:
Definitely hard code namespace should be fixed. The fact istiod is here makes me think this code path isn't even exercised?

Reply:
It is excercised with the 'standalone' install. 

We should probably add an istiod service if istiod is enabled even in the helm installer - we want to migrate to this name in 1.6 for the deployment as well.